t/t5516-fetch-push.sh

```diff
@@ -1712,4 +1712,23 @@ test_expect_success 'updateInstead with push-to-checkout hook' '
 	)
 '
 
+test_expect_success 'denyCurrentBranch and worktrees' '
+    git worktree add new-wt &&
+	git clone . cloned &&
+	test_commit -C cloned first &&
+	test_config receive.denyCurrentBranch refuse &&
+	test_must_fail git -C cloned push origin HEAD:new-wt &&
+	test_config receive.denyCurrentBranch updateInstead &&
+	git -C cloned push origin HEAD:new-wt &&
+	test_must_fail git -C cloned push --delete origin new-wt
+'
+
+test_expect_failure 'denyCurrentBranch and unborn branch' '
+	git init repo &&
+	git clone repo cloned-repo &&
+	test_commit -C cloned-repo first &&
```

Comment:
Why not just use `HEAD`? At this stage, the current directory has a fully initialized repository already, and at least a couple commits, too. `git push repo HEAD:master` should do the trick.

Reply:
Also, shouldn't this `test_expect_failure` be changed to `test_expect_success` in the commit that fixes that bug?

Reply:
> Why not just use `HEAD`? At this stage, the current directory has a fully initialized repository already, and at least a couple commits, too. `git push repo HEAD:master` should do the trick.

I come up with
```
test_expect_failure 'denyCurrentBranch and unborn branch' '
	git init unborn &&
	test_config -C unborn receive.denyCurrentBranch refuse &&
	test_must_fail git push unborn HEAD:master
'
```
I have few uncertainities regarding this bug.
Test is also passing without this patch applied.

Reply:
>  Test is also passing without this patch applied.

Oh? What is the error message, then? The `test_must_fail` idiom is very fraught with peril, as we cannot really say _how_ it failed. Maybe you can redirect `2> err` and then also verify with a `test_i18ngrep` that it displays the _expected_ error?

Reply:
```
remote: error: refusing to update checked out branch: refs/heads/master        
To unborn
 ! [remote rejected] HEAD -> master (branch is currently checked out)
error: failed to push some refs to 'unborn'
```
same output with and without this patch

Reply:
`git push unborn HEAD:master` must pass without this patch applied? if bug exists?

Reply:
Aha! So our analysis was wrong. I thought you had verified that `refs_resolve_unsafe()` fails to resolve `HEAD` if there are no commits yet? I guess that this analysis has been incorrect, or at least incomplete. For this PR to go forward, I think it must be made complete. In particular, we will need to figure out why the changes to t5509 are required. Once we figured out that correct root cause, we can determine what regression test to add for the incidental bug fix.

Reply:
I'm on it.